src/Avalonia.Base/Media/Fonts/UnmanagedFontMemory.cs

```diff
@@ -0,0 +1,407 @@
+        /// data.</returns>
+        private static UnmanagedFontMemory CreateFromBytes(ReadOnlySpan<byte> data)
+        {
+            var len = data.Length;
+            var ptr = Marshal.AllocHGlobal(len);
+
+            try
+            {
+                if (len > 0)
+                {
+                    unsafe
+                    {
+                        fixed (byte* src = &MemoryMarshal.GetReference(data))
+                        {
+                            Buffer.MemoryCopy(src, (void*)ptr, len, len);
+                        }
+                    }
+                }
+
+                return new UnmanagedFontMemory(ptr, len);
+            }
+            catch
+            {
+                Marshal.FreeHGlobal(ptr);
+                throw;
+            }
+        }
+
+        // Implement MemoryManager<byte> members on the owner
+        public override Span<byte> GetSpan()
+        {
+            _lock.EnterReadLock();
+
+            try
+            {
+                if (_disposed)
+                {
+                    throw new ObjectDisposedException(nameof(UnmanagedFontMemory));
+                }
+
+                if (_ptr == IntPtr.Zero || _length <= 0)
+                {
+                    return Span<byte>.Empty;
+                }
+
+                unsafe
+                {
+                    return new Span<byte>((void*)_ptr.ToPointer(), _length);
+                }
+            }
+            finally
+            {
+                _lock.ExitReadLock();
+            }
+        }
+
+        public override MemoryHandle Pin(int elementIndex = 0)
+        {
+            if (elementIndex < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(elementIndex));
+            }
+
+            // Increment pin count first to prevent dispose racing with pin.
+            Interlocked.Increment(ref _pinCount);
+
+            // Validate state under lock
+            _lock.EnterReadLock();
+
+            try
+            {
+                if (_disposed)
+                {
+                    throw new ObjectDisposedException(nameof(UnmanagedFontMemory));
+                }
+
+                if (_ptr == IntPtr.Zero || _length == 0)
+                {
+                    return new MemoryHandle();
+                }
+
+                if (elementIndex > _length)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(elementIndex));
+                }
+
+                unsafe
+                {
+                    var p = (byte*)_ptr.ToPointer() + elementIndex;
+                    return new MemoryHandle(p);
+                }
+            }
+            finally
+            {
+                _lock.ExitReadLock();
+            }
+        }
+
+        public override void Unpin()
+        {
+            // Decrement pin count
+            Interlocked.Decrement(ref _pinCount);
+        }
+
+        public void Dispose()
```

Comment:
`IDisposable.Dispose` is already implemented by the base class, this can be removed.